src/util/translation.h

```diff
@@ -47,13 +49,41 @@ inline bilingual_str operator+(bilingual_str lhs, const bilingual_str& rhs)
     return lhs;
 }
 
+namespace util {
+/**
+ * Translation function.
+ * If no translation function is set, simply return the input.
+ */
+inline std::string translate(const char* lit)
+{
+    return G_TRANSLATION_FUN ? G_TRANSLATION_FUN(lit) : lit;
+}
+
+struct Translatable {
+    const char* const lit;
+    consteval Translatable(const char* str) : lit{str} { assert(lit); }
+    std::string translate() const { return util::translate(lit); }
+    operator bilingual_str() const { return {lit, translate()}; }
+};
+
+template <unsigned num_params>
+struct bilingual_fmt {
```

Comment:
nit: Use Consistent naming convention in newer code as recommended by developer-notes?
```suggestion
struct BilingualFmt {
```